src/tests/formatter.rs

```diff
@@ -0,0 +1,33 @@
+use std::{env, fs::{self, Permissions}, os::unix::fs::PermissionsExt};
+
+use crate::modules::formatter::BashFormatter;
+
+fn create_fake_binary(fmt: BashFormatter) {
+    let body = if cfg!(unix) {
+        "#!/usr/bin/env bash\nexit 0"
+    } else {
+        panic!("this test is not available for non-unix platforms")
+    };
+
+    let name: String = fmt.as_cmd();
+    
+    fs::write(&name, body).expect("Couldn't write fake script");
+    fs::set_permissions(&name, Permissions::from_mode(0o755)).expect("Couldn't set perms for fake script");
+}
+
+#[test]
+fn all_exist() {
+    let path = env::var("PATH").expect("Cannot get $PATH");
+
+    env::set_var("PATH", format!("{path}:./")); // temporary unset to ensure that shfmt exists in $PATH
+    let fmts = BashFormatter::get_all();
+    for fmt in fmts {
+        create_fake_binary(fmt);
+        assert_eq!(fmt.is_available(), true);
+        assert_eq!(BashFormatter::get_available().is_some(), true);
+        fs::remove_file(fmt.as_cmd::<String>()).expect("Couldn't remove formatter's fake binary");
+    }
+
+    env::set_var("PATH", &path);
+    assert_eq!(env::var("PATH").expect("Cannot get $PATH"), path);
+}
```

Comment:
You're not testing if the formmating actually works. I know it's optional, but we still need to make sure that it works and get notified if formatting has changed.

I think it will work fine if you add formatters to testing CI deps, and manually control if they are in path var in tests.

Reply:
Will slowdown a bit the CI to test always also for shfmt if works.

It is something worth it as we are sure that it is a stable tool?

Reply:
With cache it should be fine I think. Could you test how long it would take?

And I don't think that we need to care about CI speed right now, we don't do releases like some company for SaaS with many dev teams. We can take things little slower and optimize later

Reply:
Well CI is executed at every commit in a PR to check if cargo test works....

Reply:
I agree with both of you. If anyone of you have time to do this then sure let's write some tests for it